### PR TITLE
feat(vector,helpers): add more vector functions, add helper functions

### PR DESCRIPTION
**Pull request by straker**
Adds vector functions
* subtract
* scale
* normalize
* dot
* length
* distance
* angle

Adds helper functions
* degToRad
* radToDeg
* randInt

### CODE REVIEW

File: src/vector.js [hunk 1]
```diff
@@ -26,26 +27,147 @@ class Vector {
...
+  /**
+   * Calculate the normalized value of the current vector. Requires the Vector (length)[/api/vector#length] function.
+   * @memberof Vector
+   * @function normalize
+   *
+   * @returns {Vector} a new Vector instance.
+   */
+  // @see https://github.com/jed/140bytes/wiki/Byte-saving-techniques#use-placeholder-arguments-instead-of-var
+  normalize(length = this.length()) {
+    return new Vector(
```
Comment (Rayne) [inline src/vector.js]: I always wonder if we should treat null vectors differently. A null vector would produce an `(NaN, NaN)` vector. I assume that this problem is known and the code is intentional. My math library is essentially doing the same. :-)

File: src/vector.js [hunk 2]
```diff
@@ -26,26 +27,147 @@ class Vector {
...
+  // TODO: function name due to livingcss error tags:107 (overriding length property of an array causes an error)
+  /**
+   * Calculate the length (magnitude) of the Vector.
+   * @memberof Vector
+   * @function lengthh
+   *
+   * @returns {Number} The length of the vector.
+   */
+  length() {
+    return (this.x * this.x + this.y * this.y) ** 0.5;
```
Comment (Rayne) [inline src/vector.js]: Wouldn't `Math.hypot()` safe some bytes before and after optimization? Does the compiler recognize the hypotenuse function?

```
(this.x * this.x + this.y * this.y)
Math.hypot(this.x, this.y)
```

```
Math.sqrt(this.x * this.x + this.y * this.y)
Math.sqrt(Math.hypot(this.x, this.y))
Math.hypot(this.x, this.y) ** .5
```

Comment (straker) [reply src/vector.js]: You know what, I never knew there was a `Math.hypot` function lol.

File: src/vector.js (hunk 1 above)
Comment (straker) [reply src/vector.js]: I've mostly opted out of error checking in favor or smaller size. There's sometimes I do it if when functionality is required to work (such as in Pool).

**Pull request merged**